extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ConfigBuildStep.java

```diff
@@ -127,12 +124,21 @@ AutoInjectAnnotationBuildItem autoInjectConfigProperty() {
     @BuildStep
     @Record(RUNTIME_INIT)
     void validateConfigProperties(ConfigRecorder recorder, List<ConfigPropertyBuildItem> configProperties,
-            BeanContainerBuildItem beanContainer) {
+            BeanContainerBuildItem beanContainer, BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
         // IMPL NOTE: we do depend on BeanContainerBuildItem to make sure that the BeanDeploymentValidator finished its processing
 
+        // the non-primitive types need to be registered for reflection since Class.forName is used at runtime to load the class
+        for (ConfigPropertyBuildItem item : configProperties) {
+            Type requiredType = item.getPropertyType();
+            String propertyType = requiredType.name().toString();
+            if (requiredType.kind() != Kind.ARRAY && requiredType.kind() != Kind.PRIMITIVE) {
```

Comment:
If it's an array, you don't need to register the array element type?

Reply:
I just moved what already existed from the vanilla `@ConfigProperty` basically, but I do see your point. On the other hand we haven't had any reports of it failing so I would say it's probably OK.

Reply:
Or nobody uses arrays :).

Reply:
:laughing: